app/src/org/commcare/activities/connect/ConnectIdBiometricConfigActivity.java

```diff
@@ -116,34 +128,92 @@ public void updateState(BiometricsHelper.ConfigurationStatus fingerprintStatus,
     }
 
     public void handleFingerprintButton() {
-        BiometricsHelper.ConfigurationStatus fingerprint = BiometricsHelper.checkFingerprintStatus(this,
-                biometricManager);
-        if (fingerprint == BiometricsHelper.ConfigurationStatus.Configured) {
-            finish(true, false);
-        } else if (!BiometricsHelper.configureFingerprint(this)) {
-            finish(true, true);
+
+        if (BiometricsHelper.isFingerprintConfigured(this, biometricManager)) {
```

Comment:
This logic should only handle fingerprint. Also, it needs to call BiometricsHelper.configureFingerprint if the fingerprint isn't configured yet (that function sends the user to the Android fingerprint config).